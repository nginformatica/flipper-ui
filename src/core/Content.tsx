import React, { FC } from 'react'
import styled from 'styled-components'
import { DefaultProps } from './types'

const StyledContent = styled.main`
    flex: 1;
    transition: all 500ms ease;
`

const Content: FC<DefaultProps> = ({
    children,
    padding,
    margin,
    style = {},
    ...otherProps
}) =>
    <StyledContent
        { ...otherProps }
        style={ { padding, margin, ...style } }>
        { children }
    </StyledContent>

export default Content
