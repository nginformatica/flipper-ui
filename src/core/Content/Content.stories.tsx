import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import Content from '.'

export default {
    title: 'Content',
    component: Content
} as ComponentMeta<typeof Content>

const Template: ComponentStory<typeof Content> = args => <Content {...args} />

export const Default = Template.bind({})
Default.args = {
    children: 'I am used to show Content'
}
