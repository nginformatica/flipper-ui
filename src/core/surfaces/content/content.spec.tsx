import React from 'react'
import { render, screen } from '@testing-library/react'
import Content from '.'

describe('Content', () => {
    it('should render', () => {
        render(<Content data-testid='container' />)

        const container = screen.getByTestId('container')

        expect(container).toBeDefined()
    })

    it('should render children', () => {
        render(<Content data-testid='container'>I am a Content</Content>)

        const label = screen.getByText('I am a Content')

        expect(label).toBeDefined()
    })

    it('should match snapshot', () => {
        const { container } = render(
            <Content data-testid='container'>I am a Content</Content>
        )

        expect(container).toMatchSnapshot()
    })
})
