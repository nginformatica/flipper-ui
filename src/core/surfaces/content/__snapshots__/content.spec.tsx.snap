// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content should match snapshot 1`] = `
<div>
  <main
    class="sc-bdnyFh cdroBp"
    data-testid="container"
    style="flex: 1; transition: all 500ms ease;"
  >
    I am a Content
  </main>
</div>
`;
